microscopium/screens/cellomics.py

```diff
@@ -253,29 +264,29 @@ def cellomics_semantic_filename(fn):
     -------
     semantic : collections.OrderedDict {string: string}
         A dictionary mapping the different components of the filename.
-
-    Examples
-    --------
-    >>> fn = ('MFGTMP_140206180002_A01f00d0.TIF')
-    >>> d = cellomics_semantic_filename(fn)
-    >>> d
-    OrderedDict([('directory', ''), ('prefix', 'MFGTMP'), ('plate', 140206180002), ('well', 'A01'), ('field', 0), ('channel', 0), ('suffix', 'TIF')])
     """
     keys = ['directory', 'prefix', 'plate', 'well', 'field', 'channel', 'suffix']
+
     directory, fn = os.path.split(fn)
-    filename, suffix = fn.split('.')[0], '.'.join(fn.split('.')[1:])
+    split_fn = fn.split('_')
 
-    # ignore '_stitched' tag
-    split_fn = filename.split('_')
-    if len(split_fn) == 4:
-        split_fn = split_fn[:-1]
+    if(split_fn[-1].startswith('stitch')):
```

Comment:
This is sooooo messy! Can you add comments explaining these two code paths and also how each of the splits works? And I would leave a doctest there, because the function is just too convoluted without it.